Description: Make documentation build works outside of git repository
 Current documentation build requires git and curl to get some stuff from the internet
 This patch aims to delete those dependencies, forcing git branch to master,
 and using Debian provided plantuml instead of downloading it from apache mirror
Author: Jean Baptiste Favre <debian@jbfavre.org>
Origin: other
Last-Update: 2019-01-03
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/doc/ext/traffic-server.py
+++ b/doc/ext/traffic-server.py
@@ -380,7 +380,7 @@ with open('../configure.ac', 'r') as f:
     autoconf_version = '.'.join(match.group(1).split('.', 2)[:2] + ['x'])
 
 # get the current branch the local repository is on
-git_branch = subprocess.check_output(['git', 'rev-parse', '--abbrev-ref', 'HEAD'])
+git_branch = "master"
 
 
 def make_github_link(name, rawtext, text, lineno, inliner, options={}, content=[]):
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -61,7 +61,7 @@ I18NSPHINXOPTS  = $(SPHINXOPTS)
 # The PAPER setting variables requires recursive make variable expansion, which automake
 # detects as non-portable. We bounce this through a shell script and do the expansion there.
 SBUILD = PAPEROPT_a4="$(PAPEROPT_a4)" PAPEROPT_letter="$(PAPEROPT_letter)" PAPER="$(PAPER)" \
-	PLANTUML_JAR="$(shell ext/plantuml_fetch.sh | tail -1)" \
+	PLANTUML_JAR="/usr/share/plantuml/plantuml.jar" \
 	$(srcdir)/sbuild $(SPHINXBUILD) \
 	-c $(srcdir) \
 	$(ALLSPHINXOPTS)
--- a/doc/uml/Makefile.am
+++ b/doc/uml/Makefile.am
@@ -18,7 +18,7 @@
 
 if BUILD_DOCS
 images := $(patsubst %.uml,images/%.svg,$(wildcard *.uml))
-PLANTUML_JAR := $(shell ../ext/plantuml_fetch.sh | tail -1)
+PLANTUML_JAR := /usr/share/plantuml/plantuml.jar
 
 all-am: jar-check $(images)
 endif
@@ -39,7 +39,7 @@ latex: all-am
 man: all-am
 
 images/%.svg : %.uml
-	$(JAVA) -jar $(PLANTUML_JAR) -o images -tsvg $<
+	$(JAVA) -jar $(PLANTUML_JAR) -graphvizdot /usr/bin/dot -o images -tsvg $<
 
 clean-local:
 	rm -f images/*.svg
--- a/doc/manpages.py
+++ b/doc/manpages.py
@@ -33,6 +33,9 @@ man_pages = [
     ('appendices/command-line/traffic_top.en', 'traffic_top', u'Display Traffic Server statistics', None, '1'),
     ('appendices/command-line/tsxs.en', 'tsxs', u'Traffic Server plugin tool', None, '1'),
     ('appendices/command-line/traffic_via.en', 'traffic_via', u'Traffic Server Via header decoder', None, '1'),
+    ('appendices/command-line/traffic_layout.en', 'traffic_layout', u'Traffic Server sandbox management tool', None, '1'),
+    ('appendices/command-line/traffic_cache_tool.en', 'traffic_cache_tool', u'Traffic Server cache management tool', None, '1'),
+    ('appendices/command-line/traffic_wccp.en', 'traffic_wccp', u'Traffic Server WCCP client', None, '1'),
 
     ('admin-guide/files/cache.config.en', 'cache.config', u'Traffic Server cache configuration file', None, '5'),
     ('admin-guide/files/hosting.config.en', 'hosting.config', u'Traffic Server domain hosting configuration file', None, '5'),
